src/core/ext/filters/client_channel/lb_policy/xds/cds.cc

```diff
@@ -107,13 +136,54 @@ class CdsLb : public LoadBalancingPolicy {
   // Note that this is not owned, so this pointer must never be derefernced.
   ClusterWatcher* cluster_watcher_ = nullptr;
 
+  absl::optional<XdsApi::CdsUpdate> cds_update_;
+
   // Child LB policy.
   OrphanablePtr<LoadBalancingPolicy> child_policy_;
 
   // Internal state.
   bool shutting_down_ = false;
 };
 
+//
+// CdsLb::LoadReportingPicker
+//
+
+CdsLb::LoadReportingPicker::LoadReportingPicker(
+    std::unique_ptr<SubchannelPicker> child_picker)
+    : child_picker_(std::move(child_picker)) {}
+
+CdsLb::PickResult CdsLb::LoadReportingPicker::Pick(PickArgs args) {
+  PickResult result = child_picker_->Pick(args);
+  // TODO(roth): We should ideally also record failures here in the case where
+  // a pick fails.  This is challenging, because we don't know which
+  // picks are for wait_for_ready RPCs or how many times we'll return a
+  // failure for the same wait_for_ready RPC.
+  if (result.type == result.PICK_COMPLETE && result.subchannel != nullptr) {
+    auto* subchannel_wrapper =
+        static_cast<StatsSubchannelWrapper*>(result.subchannel.get());
+    // Handle load reporting.
+    XdsClusterLocalityStats* locality_stats =
+        subchannel_wrapper->locality_stats();
+    // Record a call started.
+    locality_stats->AddCallStarted();
+    // Intercept the recv_trailing_metadata op to record call completion.
+    locality_stats->Ref(DEBUG_LOCATION, "LocalityStats+call").release();
+    result.recv_trailing_metadata_ready =
```

Comment:
For my circuit breaking PR, EDS is on top of Locality, so EDs need to call Locality's receive_trailing_metadat_ready and then EDS'.

Now CDS(Locality) is on top of EDS, so we would need to call EDS' first and then this?  Is that correct?  

I can make this change if your PR goes in first.

Reply:
Good catch!  Fixed.